### PR TITLE
Deprecate StaticCrsGraph

### PR DESCRIPTION
**Pull request by dalg24**
Transitioning it to KokkosKernels, in the KokkosSparse:: namespace

Compagnon to kokkos/kokkos-kernels#2419

### CODE REVIEW

File: containers/unit_tests/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@ foreach(Tag Threads;Serial;OpenMP;HPX;Cuda;HIP;SYCL)
       Vector
       ViewCtorPropEmbeddedDim
     )
-      if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4 AND Name STREQUAL "Vector")
+      if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4 AND NOT Name IN_LIST "Vector,StaticCrsGraph" )
```
Comment (masterleinad) [inline containers/unit_tests/CMakeLists.txt:31]:
```suggestion
      if(NOT Kokkos_ENABLE_DEPRECATED_CODE_4 AND NOT Name IN_LIST "Vector,StaticCrsGraph")
```

File: containers/unit_tests/CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +28,7 @@ foreach(Tag Threads;Serial;OpenMP;HPX;Cuda;HIP;SYCL)
...
         continue() # skip Kokkos::vector test if deprecated code 4 is not enabled
```
Comment (JBludau) [inline containers/unit_tests/CMakeLists.txt:32]:
```suggestion
        continue() # skip Kokkos::vector and/or StaticCrsGraph test if deprecated code 4 is not enabled
```

File: containers/src/Kokkos_StaticCrsGraph.hpp [hunk 3]
```diff
@@ -21,6 +21,26 @@
...
+static auto do_not_include = emit_warning_kokkos_static_crs_graph_deprecated();
+}  // namespace
+#endif
+#else
+#error "Deprecated <Kokkos_StaticCrsGraph.hpp> header is included"
+#endif
+
+#define KOKKOS_IMPL_DEPRECATED_USE_KOKKOS_SPARSE_INSTEAD \
+  KOKKOS_DEPRECATED_WITH_COMMENT(                        \
+      "Moved to KokkosKernels into KokkosSparse namespace")
```
Comment (JBludau) [inline containers/src/Kokkos_StaticCrsGraph.hpp:42]:
```suggestion
      "Moved to KokkosKernels::KokkosSparse namespace")
```
nitpicking but I think it is easier to read

**Review by masterleinad — COMMENTED** [review]
You'll also have to fix the `headers_self_contained` check. Otherwise, this looks OK.

**Review by JBludau — COMMENTED** [review]
minor things

File: containers/src/Kokkos_StaticCrsGraph.hpp (hunk 3 above)
Comment (dalg24) [reply containers/src/Kokkos_StaticCrsGraph.hpp:42]: No they actually have KokkosSparse as a namespace and I wasn't sure how to convey if you were only using only  Kokkos Core you are out of luck and need to add a dependency on Kernels

File: containers/src/Kokkos_StaticCrsGraph.hpp [hunk 4]
```diff
@@ -21,6 +21,22 @@
 #define KOKKOS_IMPL_PUBLIC_INCLUDE_NOTDEFINED_STATICCRSGRAPH
 #endif
 
+#include <Kokkos_Macros.hpp>
+
+#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
+#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+namespace {
+[[deprecated("Deprecated <Kokkos_StaticCrsGraph.hpp> header is included")]] int
```
Comment (tcclevenger) [inline containers/src/Kokkos_StaticCrsGraph.hpp:29]: Why not use `KOKKOS_DEPRECATED_WITH_COMMENT()` macro if we are including the macros?

Comment (masterleinad) [reply containers/src/Kokkos_StaticCrsGraph.hpp:29]: The symbol should always be deprecated when it's defined at all.

**Review by JBludau — APPROVED** [review]
lgtm

**Comment by dalg24** [comment]
Failures unrelated

**Pull request merged**